cleo_plugins/FileSystemOperations/FileUtils.cpp

```diff
@@ -84,20 +85,57 @@ bool File::flush(DWORD handle)
 
 DWORD File::open(const char* filename, const char* mode, bool legacy)
 {
+	char modeParsed[8] = { 0 };
+
+	// validate the mode argument
+	bool modeValid = false;
+	bool modeBin = false;
+	auto modeLen = mode != nullptr ? strlen(mode) : 0;
+	if (modeLen > 0 && modeLen < (sizeof(modeParsed) - 1)) // keep space for extra binary mode char
+	{
+		modeValid = true;
+
+		size_t i = 0;
+		while (i < modeLen)
+		{
+			auto& ch = mode[i];
+			if (ch == '+' || ch == 'a' || ch == 'b' || ch == 'r' || ch == 'w')
+			{
+				modeParsed[i] = ch;
+				modeBin = modeBin || ch == 'b';
+			}
+			else
+			{
+				modeValid = false;
+				break; // invalid character
+			}
+
+			i++;
+		}
+
+		if (!modeBin) modeParsed[i] = 'b'; // always open as binary mode, as text mode does not behave as expected in multiple scenarios
```

Comment:
can you explain these scenarios?

Reply:
Generally text mode is not well documented and many stream related functions has undefined behavior. For example `ftell` returns invalid values.

Reply:
Updated.